example_test.go

```diff
@@ -42,11 +43,10 @@ func ExampleNfqueue_Register() {
 	}
 
 	// Register your function to listen on nflqueue queue 100
-	err = nf.Register(ctx, fn)
```

Comment:
Replacing the deprecated function `Register()` with `RegisterWithErrorFunc()` in the example is a good idea. But we should keep the error handling.